Framework/CurveFitting/src/Algorithms/Fit.cpp

```diff
@@ -334,10 +266,99 @@ void Fit::execConcrete() {
+      m_domainCreator->createOutputWorkspace(baseName, m_function,
+                                             m_costFunction->getDomain(),
+                                             m_costFunction->getValues());
     }
   }
+}
+
+/** Executes the algorithm
+*
+*  @throw runtime_error Thrown if algorithm cannot execute
+*/
+void Fit::execConcrete() {
+
+  std::string ties = getPropertyValue("Ties");
+  if (!ties.empty()) {
+    m_function->addTies(ties);
+  }
+  std::string contstraints = getPropertyValue("Constraints");
+  if (!contstraints.empty()) {
+    m_function->addConstraints(contstraints);
+  }
+
+  // Try to retrieve optional properties
+  int intMaxIterations = getProperty("MaxIterations");
+  const size_t maxIterations = static_cast<size_t>(intMaxIterations);
+
+  // Get the minimizer
+  initializeMinimizer(maxIterations);
+
+  const int64_t nsteps = maxIterations * m_function->estimateNoProgressCalls();
+  API::Progress prog(this, 0.0, 1.0, nsteps);
+  m_function->setProgressReporter(&prog);
+
+  // do the fitting until success or iteration limit is reached
+  size_t iter = 0;
+  bool success = false;
+  bool isFinished = false;
+  std::string errorString;
+  g_log.debug("Starting minimizer iteration\n");
+  while (iter < maxIterations) {
+    g_log.debug() << "Starting iteration " << iter << "\n";
+    try {
+      // Perform a single iteration. isFinished is set when minimizer wants to
+      // quit.
+      m_function->iterationStarting();
+      isFinished = !m_minimizer->iterate(iter);
+      m_function->iterationFinished();
+    } catch (Kernel::Exception::FitSizeWarning &) {
+      // This is an attempt to recover after the function changes its number of
+      // parameters
+      // or ties during the iteration.
+      if (auto cf = dynamic_cast<API::CompositeFunction *>(m_function.get())) {
+        // Make sure the composite function is valid.
+        cf->checkFunction();
+      }
+      // Re-create the cost function and minimizer.
+      initializeMinimizer(maxIterations - iter);
+    }
+
+    prog.report();
+
+    if (isFinished) {
+      // It was the last iteration. Set the error string and break out of the
+      // loop.
+      errorString = m_minimizer->getError();
+      g_log.debug() << "Iteration stopped. Minimizer status string="
+                    << errorString << "\n";
+
+      success = errorString.empty() || errorString == "success";
+      if (success) {
+        errorString = "success";
+      }
+      break;
+    }
+    ++iter;
+  }
+  g_log.debug() << "Number of minimizer iterations=" << iter << "\n";
+
+  m_minimizer->finalize();
+
+  if (iter >= maxIterations) {
+    if (!errorString.empty()) {
+      errorString += '\n';
+    }
+    errorString += "Failed to converge after " + std::to_string(maxIterations) +
+                   " iterations.";
+  }
+
+  // return the status flag
+  setPropertyValue("OutputStatus", errorString);
+
+  // fit ended, creating output
+  createOutput();
```

Comment:
This method is a little long. Perhaps we could break it up a little. Some suggestions:
 - The while loop could be in a separate method (`doFit` or something) taking a `maxIterations` param and returning the number of iterations performed (`iter`). The rest of this code is really just preparing for the fit.
 - Move the `errorString` handling out of the while loop and combine it with the error string handling outside the loop at line 349. This could then maybe be made into another method  (`getOutputStatus` or something).